File: spec/spaceSpec.js
'use strict';

describe("Space", function () {

  var space

  beforeEach(function () {
    space = new Space("London Flat", "Beautiful place to stay", 5000);
  })

  describe("on initialization", function () {
    it('creates a new space with a name', function () {
      expect(space.name).toEqual("London Flat")
    })

    it('creates a new space with a description', function () {
      expect(space.description).toEqual("Beautiful place to stay")
    })

    it('creates a new space with a price per night', function () {
      expect(space.ppn).toEqual(5000)
    })
  })

  describe('so that a user can indicate the dates a space is unavailable', function () {
    it('a range of dates can be stored for when the space is not available', function () {
      let date = new Date("2020-02-28")
      space.addDateUnavailable(date)
      expect(space.datesUnavailable[0]).toEqual(date)
    })
  })
})

File: spec/space_spec.rb
require 'date'

describe Space do

  describe 'initialization' do

    it 'starts with a name' do
      expect(subject.name).to eq "London"
    end

    it 'starts with a description' do
      expect(subject.description).to eq "Beautiful place to stay"
    end

    it 'starts with a price per night' do
      expect(subject.ppn).to eq 5000
    end

    it 'starts with an empty dates unavailable array' do
      expect(subject.dates_unavailable).to eq []
    end
  end

  describe 'add dates space is unavailable' do

    it 'dates_unavailable array is not empty when date added' do
      date = Date.new(2019,12,20)
      subject.add_date_unavailable(date: date)
      expect(subject.dates_unavailable).not_to be_empty
    end

    it 'pushes the date into the dates_unavailable array' do
      date = Date.new(2019,12,20)
      subject.add_date_unavailable(date: date)
      expect(subject.dates_unavailable).to include date
    end
  end
end
